### ISSUE TITLE
`cleanBeforeInit` function throws uncaught error preventing WalletConnect QR code from being displayed

### ISSUE THREAD

**Issue by nop33** [alephium/alephium-web3#410]
This happens on the Ayin website which (as I was informed today) uses our SDK version `0.45.0`. 

Here's a video walkthrough of the error:

https://github.com/user-attachments/assets/e9a4242b-a780-45af-b2ab-df09ea52f76b

To reproduce you can delete the indexed DB:

![image](https://github.com/user-attachments/assets/0df18a54-c3ce-4011-9013-21b02f74f1ce)

**Comment by polarker** [comment]
Nice. @Lbqds Please help us investigate

**Comment by Lbqds** [comment]
The root cause of this issue is in this function: https://github.com/jakearchibald/idb-keyval/blob/6695b178d3e1a6a6252d6d72bf8b1830253aadc1/src/index.ts#L12

In normal cases, when the database does not exist, `const request = indexedDB.open(dbName)` is called to create the database, where `dbName` is `WALLET_CONNECT_V2_INDEXED_DB`. Then the `keyvaluestorage` object store is created in the `onupgradeneeded` callback.

The issue with the ayin app is that after calling `indexedDB.open('WALLET_CONNECT_V2_INDEXED_DB')`, the `onupgradeneeded` callback is not triggered, meaning the `keyvaluestorage` object store is not created, leading to the error: `One of the specified object stores was not found`.

 I’m not sure why this happens, I did the following tests:

* alephium-dex + sdk v1.5.1
* bridge-ui + sdk v1.5.1
* alephium-dex + sdk v0.45.0 (which is the version ayin app is using)

None of the above tests has this issue, after deleting database `WALLET_CONNECT_V2_INDEXED_DB`, clicking the `connect` button or refreshing the page correctly creates the database and the `keyvaluestorage` object store.

And this issue doesn’t seem to be related to `cleanBeforeInit`, because walletconnect also needs to create the `keyvaluestorage` within `SignClient.init`. I’m not sure if there’s a simple way to fix this issue in sdk. Perhaps we could ask ayin to upgrade to the latest sdk version and see if this issue still exists.

**Comment by Lbqds** [comment]
By the way, since [`migrate`](https://github.com/WalletConnect/walletconnect-utils/blob/dccc9250d018ce687b8ee6457c9b6525eb98f773/misc/keyvaluestorage/src/browser/index.ts#L15) is called without await, we can't directly catch this exception within the sdk.

**Comment by polarker** [comment]
It makes sense to upgrade to the latest SDK version. At least for wallet connect provider

**Comment by nop33** [comment]
Yes, I've already told them since yesterday to first update to latest SDK to see if the problem gets resolved.

**Comment by nop33** [comment]
The Ayin team has shared a testing link where they updated to the latest SDK (`v1.5.2`) and the same issue is observed.

**Comment by nop33** [comment]
I discovered that when clearing all cache data solves the problem for Ayin. Yet, this is not necessary for all other dApps.


https://github.com/user-attachments/assets/3b8cad08-c9d3-4bb9-a0d8-71120658ae7e
